x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
+   certificate. You must also specify the `--ca-cert` parameter.
+====
+
+   a. Enter a name for the output file or accept the default of
+   `certificate-bundle.zip`.
+
+   b. When prompted, enter a password for each node certificate. It's good
+   practice to use a different password for each node certificate and store the
+   files in a secure location.
+
+   c. Decompress the `certificate-bundle.zip` file. The output contains a
+   directory for each of your nodes with a `<node-name>.p12` file.
+
+   d. Rename the `<node-name>.p12` file to match the existing file name
+   specified by the `xpack.security.transport.ssl.keystore.path` variable in 
+   the `elasticsearch.yml` file on each node.
+
+. [[start-rolling-restart-newca,step 3]]On the current node in your cluster where
+you're updating the keystore, start a
+<<restart-cluster-rolling,rolling restart>>.
++
+Stop at the step indicating *Perform any needed changes*, and then proceed to
+the next step in this procedure.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `<node-name>.p12`.
++
+If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename so that {es} doesn't attempt to reload the file
+before you update the password.
+
+. If you needed to save the new keystore with a new filename, update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore.
+For example:
++
+[source,yaml]
+----
+xpack.security.transport.ssl.keystore.path: config/<node-name>.p12
+xpack.security.transport.ssl.keystore.type: PKCS12
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. Start the node where you updated the keystore, and then complete the remaining
+steps for a <<restart-cluster-rolling,rolling restart>>.
+
+. [[verify-keystore-newca,step 7]](Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. One node at a time, complete <<start-rolling-restart-newca>> through
+<<verify-keystore-newca>> until you've updated all keystores in your cluster.
+
+. (Optional) After rotating keystores on each node in your cluster, remove the
+old CA certificate from your truststore.
++
+[source,shell]
+----
+keytool -delete -noprompt -alias <old-ca>  -keystore config/elastic-stack-ca.p12 \
+-storepass <password>
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `-alias`:: Name of the old CA certificate that you want to remove from your
+   truststore.
+====
+
+[discrete]
+[[transport-layer-newca-whatsnext]]
+==== What's next?
+Well done! You've updated the keystore for the transport layer. You can also
+<<node-certs-different-http,update the keystore for the HTTP layer>> if
+necessary. If you're not updating the keystore for the HTTP layer, then you're
+all set. 
+
+[[node-certs-different-http]]
```

Comment:
Somewhere here, early on, we need to tell the users that they can go to all the clients that connect to elasticserach ( kibana, beats,logstash,agent, language clients ) and configure them to also trust the new CA ( ca.crt ) that we generated. Then in the last steps of this section inform them that they can go back to the configuration of these clients and remove the trust old CA certificate. We can point to existing configuration. 

The issue is that if we don't mention anything and users will go through this section, they will have effectively cut off their ES cluster from anything that connect to it , and then they'd need to scramble to fix things or revert changes

Reply:
I'm adding some explanation at the beginning of this section about trusting the new CA for all clients, and adding the first two steps for importing and verifying that new CA.

Reply:
> @jkakavas, do we only need to have users import the new CA into the existing keystore for the HTTP layer  

no, the don't need to